### ISSUE TITLE
Unable to read execution logs

### ISSUE THREAD

**Issue by Vishal-Mendekar** [tesseract-ocr/tesstrain#242]
**I trained tesseract-ocr model on english numbers for 0.1 million(100000) iterations and the output I got in terminal is attached below.**

![run logs](https://user-images.githubusercontent.com/54699229/114274957-40aa7800-9a10-11eb-9e61-80731838fe72.png)

But I am not sure whether the model ran for 0.1 million iterations. Also I am unable to read the execution logs.

@jimregan @stweil @Shreeshrii

**Comment by stweil** [comment]
It looks like training stopped because the character and word error rates dropped to 0 %. Therefore it did not run 100000 iterations.

**Comment by sawankumar94** [comment]
Thank you @stweil ! 

How can one make sure that the code run for all the iterations despite the fact that character and word error rates drops to 0. ?  

How can one make sure that the code runs for all the iterations?  What changes should one make in the Makefile script to make it run for all the iterations? 

TIA.

**Comment by stweil** [comment]
Add more training data? As soon as the model is perfect (error rate 0 %) more iterations won't change it, so why would you want to run more iterations?

**Comment by sawankumar94** [comment]
We have augmented the data by adding noise, blurring, etc. initially the data size was 7k but later we made it to 200k. Therefore, I want the model to go through all the data. So that, it works well on real-life images (improper data). 

What does  **At Iteration 500/59300/59300** mean in the log file attached ? 

TIA.

**Comment by M3ssman** [comment]
You can alter the target error rate `TARGET_ERROR_RATE` for training when you call the Makefile to affect the actual duration. 
It defaults to 0.01, although this is already pretty low.

**Comment by bertsky** [comment]
> What does **At Iteration 500/59300/59300** mean in the log file attached ?

IIUC it means 59300 lines were processed, 500 of which produced a large enough error/gradient to warrant an update of the weights.

> We have augmented the data by adding noise, blurring, etc. initially the data size was 7k but later we made it to 200k. Therefore, I want the model to go through all the data. So that, it works well on real-life images (improper data).

You can either set `MAX_ITERATIONS` high enough (i.e. to some _multiple_ of the total number of lines you have), or instead set `EPOCHS` for that (which effectively calls `lstmtraining --max_iterations -$(EPOCHS)`, where the negative value is understood by Tesseract 5.0 as the number of iterations over all your files).

But mind @M3ssman's answer, too.
